component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/front/ComponentResourceImpl.java

```diff
@@ -610,6 +614,18 @@ private String normalizeCategory(final String category) {
         return category;
     }
 
+    private Map<String, String> translateMetadata(final Map<String, String> source, final ComponentBundle bundle) {
+        return source
+                .entrySet()
+                .stream()
+                .map(e -> {
+                    final String bundleKey = e.getKey().replaceAll("::", ".");
+                    e.setValue(bundle.displayName(bundleKey).orElse(e.getValue()));
```

Comment:
Is it okay to change the internal state of the in-parameter source?
I see ".orElse()" part will back us up, but still. 
From the javadoc of setValue:
> Replaces the value corresponding to this entry with the specified value (optional operation). (Writes through to the map.)